### PR TITLE
Feature/nm 105

### PR DESCRIPTION
**Pull request by NorbertVozarik**

### CODE REVIEW

File: services/utils.service.js [hunk 1]
```diff
@@ -133,6 +133,26 @@ function validateDates(startDate, endDate) {
     }
 }
 
+function validateDatesIntervals(dateIntervalsArray) {
```
Comment (tjalc) [inline services/utils.service.js:136]: I think that Tomas Mihaly already added the same function validateDateArrayParam. He did merge his PR to master today morning.  Did you merge latest changes from master to your feature branch?

Comment (NorbertVozarik) [reply services/utils.service.js:136]: ok, I will delete my function and use his

Comment (NorbertVozarik) [reply services/utils.service.js:136]: There is nothing submitted yet. I merged master to my branche but I don't see any new changes

File: services/commits.service.js [hunk 2]
```diff
@@ -71,14 +83,26 @@ function getCommitForTeam(teamName, startDate, endDate) {
...
-                endDate: endDate
+            let commitDate;
+            let result = {team: teamName};
+            let countIntervalsArray = [];
+            for (i = 0; i < dateIntervalsArray.length; i++) {
+                let countCommits = 0;
+                let countCommitsWithoutJiraReference = 0;
+                for (j = 0; j < response.length; j++) {
+                    commitDate = response[j].commit.author.date;
+                    if (commitDate >= dateIntervalsArray[i].startDate && commitDate < dateIntervalsArray[i].endDate) {
```
Comment (tjalc) [inline services/commits.service.js:94]: is it safe to compare dates like strings?
This will not work when dates in request will have different timezone than commit's date returned from githubAPI.

Lets say, there is one commit made by user on 2020-03-26T10:00:00Z(Github API return dates in UTC timezone) and we send following date interval in request dateIntervals=[{"startDate":"2020-03-26T12:00:00+0200","endDate":"2020-03-26T14:00:00+0200"} so timezone is UTC+2

So we have following case:
//comparing Strings
let startDateStr = "2020-03-26T12:00:00+02:00";
let endDateStr = "2020-03-26T14:00:00+02:00";
let commitDateStr = "2020-03-26T10:00:00Z"
console.log(commitDateStr >= startDateStr && commitDateStr < endDateStr);
> false
- So commit will not match date intervals

// comparing Dates
let startDate = Date.parse(startDateStr);
let endDate = Date.parse(endDateStr);
let commitDate = Date.parse(commitDateStr);
console.log(commitDate >= startDate && commitDate < endDate);
> true
// Commit will match date intervals

Comment (NorbertVozarik) [reply services/commits.service.js:94]: ok, I will fix that

Comment (NorbertVozarik) [reply services/commits.service.js]: fix done

File: services/impact.service.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ function getUserImpact(username, startDate, endDate) {
         })
 }
 
-function getTeamImpact(teamName, startDate, endDate){
+function getTeamImpact(teamName, dateIntervals){
```
Comment (tjalc) [inline services/impact.service.js:14]: why did you change impact.service?

Comment (NorbertVozarik) [reply services/impact.service.js:14]: I didn't change....I just pulled the changes from date-array-params branche….

Comment (tjalc) [reply services/impact.service.js:14]: hm, I think it won't work as the function body is not changed, only parameters. Could you please revert these impact.service and controller changes?
